dataplicity/portforward.py

```diff
@@ -151,7 +168,9 @@ def _connect(self):
             log.exception('error connecting')
             return False
         else:
-            log.debug("connected to %s", self.service.url)
+            if self.service:
+                log.debug("connected to %s", self.service.url)
+            _socket.setblocking(0)  # set non-blocking
```

Comment:
Why should the socket be non-blocking? Given we are using select to let us read only when there is data...

Reply:
I have no idea but this is not my code :D all I actually did was to turn log.debug into a conditional log.debug since I was having some exceptions when this code was executed and the self.service was None. the rest came from master branch

Reply:
I can't see this in master. Could this be the result of a bad merge?

Reply:
ah, ok I see that Will removed it with the last commit to master. which  means that this line came from the point when I was doing the branch. I'll remove it then

Reply:
I don't think that it was due to some merge error, as:

```
git rebase origin/master 
Current branch bugfixes-for-redirect-port is up to date.
```

so I can't really explain why this happened